src/utils/splitAtom.ts

```diff
@@ -38,7 +38,7 @@ export function splitAtom<Item, Key>(
   if (cachedAtom) {
     return cachedAtom
   }
-  type ItemAtom = PrimitiveAtom<Item> | Atom<Item>
+  type ItemAtom = PrimitiveAtom<Item | undefined> | Atom<Item | undefined>
```

Comment:
Oh, this is very unfortunate. And not correct for PrimitiveAtom, because we can't update a value with `undefined`.

Maybe we can just return `undefined` blindly.

<img width="263" alt="image" src="https://user-images.githubusercontent.com/490574/122927942-53dcab00-d3a4-11eb-86c2-742dfa4b50bf.png">

[playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAHgLhmArgWwEYFMBOBtAXRgF4Y8BGAGgCZKBmAgbgChXRJYBDE+PATibMgA)